File: 374-guess-number-higher-or-lower/guess-number-higher-or-lower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if num is higher than the picked number
#          1 if num is lower than the picked number
#          otherwise return 0
# def guess(num: int) -> int:

class Solution:
    def guessNumber(self, n: int) -> int:
        l = 1
        r = n
        while l <= r :
            num = (l+r)//2
            g = guess(num)
            if g == -1:
                r = num-1
            elif g == 1:
                l = num+1
            else:
                return num

File: 374-guess-number-higher-or-lower/guess-number-higher-or-lower.js
/** 
 * Forward declaration of guess API.
 * @param {number} num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * var guess = function(num) {}
 */

/**
 * @param {number} n
 * @return {number}
 */
var guessNumber = function(n) {
    let l = 1
    let r = n
    while (l<=r){
        let mid = Math.floor((l+r)/2)
        let g = guess(mid)
        if (g === -1){
            r = mid-1
        }else if (g == 1){
            l = mid+1
        }else{
            return mid
        }
    }
};